sdk/messaging/azservicebus/internal/utils/logger.go

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package utils
+
+import (
+	"fmt"
+	"sync/atomic"
+
+	azlog "github.com/Azure/azure-sdk-for-go/sdk/internal/log"
+)
+
+type Logger struct {
+	prefix *atomic.Value
```

Comment:
Is this actually used concurrently?

Reply:
It's possible, yes.  This logger is used in all locations where the Receiver (or Sender) does any opeations.

For instance, the user can do a parallel message completion like what we do in send_receive_drain.go stress test.

Reply:
Is this for tests only?

Reply:
It's for production code.

Reply:
Understood per offline discussion.